csrc/type_convert.cuh

```diff
@@ -0,0 +1,167 @@
+#pragma once
+
+#include <torch/all.h>
```

Comment:
I think including `torch/all.h` is probably overkill and pulls in a bunch of stuff we don't need.

Could you try including `torch/cuda.h` and maybe add `torch/torch.h` if that's not enough?